packages/react/focus-scope/src/FocusScope.tsx

```diff
@@ -21,10 +21,21 @@ type FocusScopeOwnProps = Polymorphic.Merge<
   Polymorphic.OwnProps<typeof Primitive>,
   {
     /**
-     * Whether focus should be trapped within the FocusScope
+     * Whether focus should wrap around the `FocusScope`.
+     * If `true`, tabbing out of the last tabbable element inside `FocusScope`
+     * will focus the first tabbable element inside and vice-versa.
      * (default: false)
      */
-    trapped?: boolean;
+    wrapped?: boolean;
+
+    /**
+     * Whether focus should be contained inside the `FocusScope`.
+     * If `true`, tabbing out of the last tabbable element inside `FocusScope`
+     * will do nothing. Also, when moving focus programatically outside of it, focus will
+     * be brought back in.
+     * (default: false)
+     */
+    contained?: boolean;
```

Comment:
I was bored and wanted to jot down a few things quickly before this is merged:

- Thoughts on `loop` instead of `wrapped`? When something "wraps" something it "encloses" it, so it sounds like this would put the focus around the whole scope (on the content). 
- Thoughts on `trap` instead of `contained`? For some reason `trap` communicates that it cannot escape a bit better to me.
- For `contained` where it says tabbing the last element will do nothing, it makes it sound like `contained` and `wrapped` cannot be used together because they contradict each other.

I think something along these lines is a bit clearer:

```suggestion
    /**
     * When `true`, loops tab focus within the focus scope. Tabbing from last item
     * will focus first tabbable and shift+tab from first item will focus last tababble.
     * @defaultValue false
     */
    loop?: boolean;

    /**
     * When `true`, focus cannot escape the focus scope via tab keypress or a programmatic focus.
     * @defaultValue false
     */
    trap?: boolean;
```

Reply:
I like the suggestions 👍 
On the naming, how do we feel about the verbs vs, past participle stuff? ie. `trap` vs `trapped`?
Also I think we need to think how these get renamed in upper layers.

Reply:
Really like these suggestions, particularly `trapped`. Past participle is my preference for the consistency.

Reply:
Just catching up with this now, what are we going for then?

- `contained` => `trapped`?
- `loop` => `?` ?

Interesting point to note also that we have a somewhat similar concept with a `loop` prop also on all the things that use roving focus (menus, tabs, radio-group, toolbar), so probably worth taking into consideration in this discussion too.

Reply:
I guess if we're going past participle then it would be `trapped` and `looped` but i'm also fine with `loop`.

Reply:
Same. Whichever feels better holistically to you.

Reply:
Oh and at the risk of bike shedding here, what's been the approach with the past participle stuff in the public apis? I favour intuitive over dogmatic consistency tbh but would we be saying `trapFocus` should become `trappedFocus` (and why is it `loop` and not `loopFocus` or `loopedFocus`?)

Anyway, shut me down if that's out of scope.

Reply:
> I favour intuitive over dogmatic consistency tbh

Same. The DOM APIs aren't consistent for DOM APIs either tbh. They have a `loop` attribute so my preference would be `trapped` and `loop`. 

> and why is it `loop` and not `loopFocus` or `loopedFocus`?

Since we're discussing props on a `FocusScope` component, it seems redundant to repeat the "focus" part. That would be like having a `cat` object with `catName`, `catAge`, `catBreed` properties otherwise.

```js
const cat = {
	catName: 'Joplin',
	catAge: 2,
	catBreed: 'Maine Coon' 
};

// vs

const cat = {
	name: 'Joplin',
	age: 2,
	breed: 'Maine Coon' 
};
```

Reply:
I'm happy with that 👍

Reply:
> Since we're discussing props on a FocusScope component, it seems redundant to repeat the "focus" part. That would be like having a cat object with catName, catAge, catBreed properties otherwise.

Yeah totally, was referring to public props on primitives eg places where we use `TrapFocus` but that's irrelevant in the context of this PR 👍